### PR TITLE
Non thermal models clean up and refactor

### PR DESCRIPTION
**Pull request by samaloney**
## PR Description

Less code, more readable code, and some performance improvements

* Remove some unused and unnecessary thin wrappers
* The old `gauss_legendre` port and the `fixed_quad_batch` are almost identical which should now be clearer the later is numerically better as the scaling operations are not compounded in the integration as they are done at the end - changed default to `fixed_quad_batch`
* Figure out source of some of the numeric constants and added comment with units
* Pull out a bunch of static constants to module level vars 
* Modernised some of the code structure and eliminated unnecessary checks
* More / improved comments
* More edge case tests
* ThickTarget ~ 1.5x and ThinTarget ~2x faster

- [ ] Still importing constants from legacy should we just switch to the astropy values at this stage or also pull out the constant function from legacy

## AI Assistance Disclosure

AI tools were used for:
- [  ] Code generation (e.g., when writing an implementation or fixing a bug)
- [x ] Test/benchmark generation
- [  ] Documentation (including examples)
- [x ] Research and understanding
- [  ] No AI tools were used

> Regardless of AI use, the human contributor remains fully responsible for correctness, design choices, licensing compatibility, and long-term maintainability.

### CODE REVIEW

File: sunkit_spex/models/physical/integrate.py
Comment (samaloney) [inline sunkit_spex/models/physical/integrate.py:1]: It much easier to see that `gauss_legendre` and ``gauss_legendre`` do almost the same thing the only difference is when the mapping to the real limits is done before `gauss_legendre` or after in `gauss_legendre` which mean the numerical error aren't folded in to the integration

**Comment by samaloney** [comment]
Just some notes I think this code is now close the limit of the possible performance without going down the Cython or numba routes which have there own complications.

The main bottle necks are `bremsstrahlung_cross_section` and `BrokenPowerLawElectronDistribution.density` which take up ~50% of the time.

Figure shows a performance comparison between main, this branch and SSWIDL function. The main reason `think_target` is faster is that it doesn't require the `.density` method.
image

File: sunkit_spex/models/physical/nonthermal.py [hunk 1]
```diff
@@ -1,33 +1,70 @@
...
 import astropy.units as u
 from astropy.modeling import FittableModel, Parameter
 
-from sunkit_spex.legacy import constants as const
-from sunkit_spex.legacy.integrate import gauss_legendre
+from sunkit_spex.legacy import constants
+from sunkit_spex.models.physical.integrate import fixed_quad_batch
 
-const = const.Constants()
+const = constants.Constants()
```
Comment (KriSun95) [inline sunkit_spex/models/physical/nonthermal.py:25]: I like the idea of having our own specific constants (e.g., from hard-coded IDL sources and now just Astropy ones). One thought I had is that it would make sense in the future to add Astropy units to these. I realised this was useful in #293

File: sunkit_spex/models/physical/nonthermal.py [hunk 2]
```diff
@@ -119,29 +148,20 @@ def __init__(
...
+        output = bremsstrahlung_thick_target(
+            (energy_centers << u.keV).value,
+            p,
+            (break_energy << u.keV).value,
+            q,
+            (low_e_cutoff << u.keV).value,
+            (high_e_cutoff << u.keV).value,
+            self.integrator,
+        )
+        output[~np.isfinite(output)] = 0
```
Comment (KriSun95) [inline sunkit_spex/models/physical/nonthermal.py:162]: This can be addressed in another PR but I think I included this to avoid weird values in the output. In #291 I think I remove this which forced me to try and track down when things weren't quite right.

**Review by KriSun95 — APPROVED** [review]

**Review by jajmitchell — APPROVED** [review]
